docs/protocol/core/v3.0.rst

```diff
@@ -101,7 +101,128 @@ TODO define constraints on node names
 Data types
 ----------
 
-TODO define core data types
+A data type describes the set of possible binary values that an array
+element may take, along with some information about how the values
+should be interpreted.
+
+This protocol defines a limited set of data types to represent Boolean
+values, integers, floating point numbers and complex numbers. Protocol
+extensions may define additional data types. All of the data types
+defined here have a fixed size, in the sense that all values require
+the same number of bytes. However, protocol extensions may define
+variable sized data types.
+
+Note that the Zarr protocol is intended to enable communication of
+data between a variety of computing environments. The native byte
+order may differ between machines used to write and read the data.
+
+Each data type is associated with an identifier, which can be used in
+metadata documents to refer to the data type. For the data types
+defined in this protocol, the identifier is a simple ASCII
+string. However, protocol extensions may use any JSON value to
+identify a data type.
+
+Core data types
+~~~~~~~~~~~~~~~
+
+.. list-table:: Data types
+    :header-rows: 1
+    * - Identifier
+      - Numerical type
+      - Size (no. bytes)
+      - Byte order
+    * - `bool`
+      - Boolean
```

Comment:
It would be good to clarify the binary encoding used for booleans, e.g., `False` -> `\x00` and `True` -> `\x01`

Reply:
Done.